src/main/java/org/ethereum/beacon/discovery/storage/LocalNodeRecordStore.java

```diff
@@ -9,26 +9,37 @@
 import org.ethereum.beacon.discovery.schema.NodeRecord;
 
 public class LocalNodeRecordStore {
+
   private volatile NodeRecord latestRecord;
   private final Bytes privateKey;
+  private final NodeRecordListener recordListener;
 
   public LocalNodeRecordStore(final NodeRecord record, final Bytes privateKey) {
```

Comment:
(nit) Might be a good idea to just have 1 constructor and require the listener argument.